### ISSUE TITLE
Windows Install

### ISSUE THREAD

**Issue by tomegouveia** [mamrhein/decimalfp#8]
Hi

Following from the issue I created a while back regarding installing in windows, I have a suggestion of a simple modification which allows the code to be installed in windows systems without having to clone the repository and install "by hand". I couldn't create a pull request so I'm raising it here.

I suggest replacing setup.py with the following

```
# coding=utf-8
"""Setup package 'decimalfp'."""

import os
import sysconfig

from setuptools import Extension, setup
import platform


if not (platform.system() == 'Windows'):
    LIBFPDEC_PATH = 'src/decimalfp/libfpdec'
    LIBFPDEC_SRC_FILES = sorted(f"{LIBFPDEC_PATH}/{fn}"
                                for fn in os.listdir(path=LIBFPDEC_PATH)
                                if fn.endswith(('.c',)))

    DEBUG = int(os.getenv("DEBUG", 0))
    cflags = sysconfig.get_config_var('CFLAGS')
    if cflags:
        extra_compile_args = cflags.split() + ["-Wall", "-Wextra"]
    else:
        extra_compile_args = ["-Wall", "-Wextra"]
    if DEBUG:
        extra_compile_args += ["-g3", "-O0", f"-DDEBUG={DEBUG}", "-UNDEBUG"]
    else:
        extra_compile_args += ["-DNDEBUG", "-O3"]

    ext_modules = [
        Extension(
            'decimalfp._cdecimalfp',
            ['src/decimalfp/_cdecimalfp.c'] + LIBFPDEC_SRC_FILES,
            include_dirs=['src/decimalfp', LIBFPDEC_PATH],
            extra_compile_args=extra_compile_args,
            # extra_link_args="",
            language='c',
            ),
        ]

with open('README.md') as file:
    long_description = file.read()

kwargs = dict(name="decimalfp",
    author="Michael Amrhein",
    author_email="michael@adrhinum.de",
    url="https://github.com/mamrhein/decimalfp",
    description="Decimal numbers with fixed-point arithmetic",
    long_description=long_description,
    long_description_content_type="text/markdown",
    package_dir={'': 'src'},
    packages=['decimalfp'],
    python_requires=">=3.7",
    license='BSD',
    keywords='fixed-point decimal number datatype',
    platforms='all',
    classifiers=[
        "Development Status :: 5 - Production/Stable",
        "Intended Audience :: Developers",
        "License :: OSI Approved :: BSD License",
        "Operating System :: OS Independent",
        "Programming Language :: Python",
        "Programming Language :: Python :: 3",
        "Programming Language :: Python :: 3.7",
        "Programming Language :: Python :: 3.8",
        "Programming Language :: Python :: 3.9",
        "Programming Language :: Python :: 3.10",
        "Programming Language :: Python :: Implementation :: CPython",
        "Programming Language :: Python :: Implementation :: PyPy",
        "Topic :: Software Development",
        "Topic :: Software Development :: Libraries",
        "Topic :: Software Development :: Libraries :: Python Modules"
        ],
    zip_safe=False)

if not (platform.system() == 'Windows'):
    kwargs['package_data'] = {'decimalfp': ['py.typed', '_cdecimalfp.pyi']}
    kwargs['ext_modules'] = ext_modules

setup(**kwargs)

```

I hope this is considered!

Best wishes
Tomé

**Comment by mamrhein** [comment]
I will enhance setup.py by checking whether the GNU C Compiler is available and make a "source-only" install if that is not the case.

**Comment by mamrhein** [comment]
Fixed with v0.12.2.
